### ISSUE TITLE
Unclear how to enable type checking after removing `--type-check`

### ISSUE THREAD

**Issue by zamiang** [palantir/tslint#3399]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
// code snippet
```

with `tslint.json` configuration:

```json
{
  "linterOptions": {
    "typeCheck": true
  }
}
```

#### Actual behavior

No type checking happens.

The latest tslint recommends removing `--type-check`. I now run `tslint --project tsconfig.json --config tslint.json`. Even though I have `typeCheck: true` in my `tslint.json`, no type checking is performed if I remove the deprecated `--type-check`.

Is there a new config I am supposed to specify beyond adding typeCheck: true?

#### Expected behavior

Type checking happens

**Comment by adidahiya** [comment]
We are removing type checking as a feature of TSLint, that's why we deprecated the CLI flag. Type checking errors should be reported by the compiler, not the linter.

**Comment by zamiang** [comment]
Ah! I was unaware of that large change. 

It seems like that change of functionality should be very clearly communicated. Am I safe to assume that type checking will continue to work at least until the next major version?

**Comment by adidahiya** [comment]
I don't consider it a huge change... we are still allowing rules to _use_ the type checker in their implementations. See here for more info https://github.com/palantir/tslint/pull/3322#issuecomment-338641406

Yeah, we can call it out more loudly in future release notes.

**Comment by nervebassmaster** [comment]
Yep but this change's *completely killed* `pre-commit` *hooks*. It seems like now it's *possible to commit broken changes*

**Comment by ajafff** [comment]
@nervebassmaster That means you are probably using the wrong tool. Use `tsc --noEmit` in your precommit hook to make sure there are no compile errors

**Comment by nervebassmaster** [comment]
[@ajafff](https://github.com/ajafff) And what is the best way to do the same if I use ts-node instead of tsc

**Comment by nervebassmaster** [comment]
I guess I see. I found the issue in my script. 
thx, **[@ajafff](https://github.com/ajafff)**

**Comment by jupl** [comment]
I understand removing type checking, but right now I need to use `--type-check` for TSLint rules that require type info. Without this option there's missing errors that are not showing up. How do I get rule violations for rules where type info is required?

**Comment by MiguelMadero** [comment]
I was considering using `tsc --noEmit` but that adds about 7 more seconds to our build time for each package, while tsling with `--type-check` is only ~0.9 seconds, I'm assuming a lot of the work is duplicated between both tools.

**Comment by jamonkko** [comment]
I understand the design decision from technical perspective to remove typechecking from tslint. However as Miguel pointed out it has negative usability effect. 

More devs will be bumping into this when they remove the deprecating `no-unused-variable` option and try to use the compiler options (`noUnusedLocals` and `noUnusedParameters`) instead.

So if I want to do both typechecking and linting using npm scripts I need to do something like:
```
"lint": "yarn typecheck && yarn tslint",
"tslint": "tslint --project tsconfig.json --config tslint.json '{src,test}/**/*.ts'",
 "typecheck": "tsc --project tsconfig.json --noEmit"
```

and then I can just:
`yarn lint`

This is slower and much more script code than before... I wouldn't be surprised if someone will write a wrapper or fork that does both.

**Comment by JoshuaKGoldberg** [comment]
See https://github.com/Microsoft/typescript-tslint-plugin for a_(n experimental, for now)_ VS Code plugin that works with typed rules.

**Comment by kasvtv** [comment]
I really hope that the maintainers listen to their community on this one.
`--type-check` is still marked as deprecated and it makes precommit hooks terribly terribly slow to run `tsc` before.

**Comment by JoshuaKGoldberg** [comment]
@kasvtv in case you missed it, TSLint is deprecated now: #4534. You'll want to switch to https://typescript-eslint.io. 

I hear your point and agree with the pain point, but at this point your best bet is to switch tools anyway.
